### ISSUE TITLE
"Go to beatmap" button inserts the beatmap set into the currently selected collection

### ISSUE THREAD

**Issue by rederick29** [ppy/osu#11325]
**Describe the bug:** Selecting a collection in the song select menu and then clicking the "go to beatmap" button in osu!direct on a map that is not in the collection makes the map appear in the collection. This does not add the map to the collection but only shows is as being in it. 

**Screenshots or videos showing encountered issue:** 

**osu!lazer version:** 2020.1225.0 

**Logs:**

[runtime.log](https://github.com/ppy/osu/files/5743436/runtime.log)
[performance.log](https://github.com/ppy/osu/files/5743437/performance.log)

**Comment by LittleEndu** [comment]
Pretty sure this is indented. Go to beatmap ignores all filters song select has

**Comment by bdach** [comment]
Yeah, that's how it will work, as presenting a beatmap will bypass filter, but it perhaps is not the most understandable UX...

**Comment by rederick29** [comment]
If it is intended, then it is misleading in my opinion, as it can make people think that the beatmap set was added to their collection. 
Proposal: If the beatmap set is not in the currently selected collection, then the song select menu should change filtering to "All beatmaps" and then show the map when the "Go to beatmap" button is clicked.

**Comment by peppy** [comment]
It is impletmented the way it is to avoid losing the user's filter.

If it's going to change to "all beatmaps" then there needs to be a way to restore previous filter settings.

**Comment by Joehuu** [comment]
Song select v2 does not do this anymore (i.e. beatmap still gets selected but isn't shown in carousel anymore).
